spec/data-model.md

```diff
@@ -0,0 +1,183 @@
+# DRAFT MessageFormat 2.0 Data Model
+
+To work with messages defined in other syntaxes than that of MessageFormat 2,
+an equivalent data model representation is also defined.
+Implementations MAY provide interfaces which allow
+for MessageFormat 2 syntax to be parsed into this representation,
+for this representation to be serialized into MessageFormat 2 syntax
+or any other syntax,
+for messages presented in this representation to be formatted,
+or for other operations to be performed on or with messages in this representation.
+
+Implementations are not required to use this data model for their internal representation of messages.
+
+To ensure compatibility across all platforms,
+this interchange data model is defined in terms of JSON-compatible values.
+While this document uses TypeScript syntax for their definition,
+the canonical and authoritative source is the `message.json`
+[JSON Schema](https://json-schema.org/) definition.
```

Comment:
@aphillips @stasm This is the paragraph that'll need to be replaced to drop the JSON schema.